app/services/hyrax/listeners/metadata_index_listener.rb

```diff
@@ -16,7 +16,17 @@ class MetadataIndexListener
       #
       # @param event [Dry::Event]
       def on_object_metadata_updated(event)
-        Hyrax.index_adapter.save(resource: event[:object])
+        Hyrax.index_adapter.save(resource:
+          Hyrax::ValkyrieIndexer.for(resource: event[:object]).resource)
```

Comment:
i'm not following this. the index adapter does this factory lookup, right? the client should only need to pass in the resource.

i feel like i might be missing something :)

Reply:
In my console testing it seemed like you had to pass the resource like this, but I'm not 100% on that so let me do more testing.